File: CourierAmexAPI/CourierAmex/Models/CashierModel.cs
namespace CourierAmex.Models
{
    public class CashierModel : BaseEntityModel<int>
    {
        public string Name { get; set; }
        public string CompanyName { get; set; }
        public string PrinterName { get; set; }
        public string IpAddress { get; set; }
        public int PortNumber { get; set; }
        public new short Status { get; set; }
        public int CompanyId { get; set; }
    }

    public class UserByPointOfSaleModel
    {
        public int CompanyId { get; set; }
        public int PointOfSaleId { get; set; }
        public string User { get; set; }
        public string UserNumber { get; set; }
    }

    public class UserCashier
    {
        public int CompanyId { get; set; }
        public int PointOfSaleId { get; set; }
        public string User { get; set; }
    }
}

File: CourierAmexClient/src/app/features/company/models/cashier.model.ts
import { BaseEntity } from "@app/models";

export interface CashierModel extends BaseEntity<number> {
  name: string;
  companyId?: number;
  companyName?: string;
  printerName?: string;
  createdAt?: string;
  portNumber?: number;
  ipAddress?: string;
}

export interface UserByPointOfSaleModel {
  companyId: number;
  pointOfSaleId: number;
  user: string;
  userNumber: string;
}


export const newCashierModel = {
  id: 0,
  name: '',
  printerName: '',
  portNumber : 0,
  ipAddress: '',
  createdAt: new Date().toISOString(),
  companyName: "New",
  status: 1,
  companyId: undefined
};
